### PR TITLE
Admins without R_ADMIN perm are sorted in another section in staff_who

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Admins without R_ADMIN perm are sorted in another section in staff_who
See the test evidence

## Why It's Good For The Game
some people think I can handle tickets

## Testing Photographs and Procedure

image

## Changelog
:cl:
admin: Admins without R_ADMIN perm are sorted in another section in staff_who, so that people will not believe coders can handle adminhelps.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
Another idea:
* Make another flag `R_NOT_ADMIN`, and give it to headcoder and coders?

File: code/modules/client/verbs/who.dm [hunk 1]
```diff
@@ -100,13 +103,30 @@
...
+				else // admins without R_ADMIN perm should be sorted in different area, so that people won't believe coders will handle ahelp
+					var/rank = "\improper [C.holder.rank]"
+					non_admin_list += "\t[C] is \a [rank]\n"
+
+		msg = "<b>Current Admins:</b>\n"
+		for(var/each in admin_list)
+			msg += each
+		if(length(non_admin_list)) // notifying the absence of non-admins has no point
+			msg += "<b>Current Non-admin staffs:</b>\n"
+			msg += "\t(Note: non-admin staffs can't handle adminhelp tickets.)\n"
```
Comment (Ivniinvi) [inline code/modules/client/verbs/who.dm]:
```suggestion
			msg += "\t(Note: non-admin staff are unable to handle adminhelp tickets.)\n"
```

**Comment by EvilDragonfiend** [comment]
thanks for the grammar correction lol

**Comment by WrillWasTaken** [comment]
Wouldn't it be better to have a coder/maintainer category? This seems unclear, mentors are technically non-admin staff who can't handle tickets too.

**Comment by EvilDragonfiend** [comment]
> Wouldn't it be better to have a coder/maintainer category? This seems unclear, mentors are technically non-admin staff who can't handle tickets too.

unlike how mentor works, maintainers get 'debug' permission from admin datum, and doing as you suggested needs greater work.

**Comment by Crossedfall** [comment]
> and doing as you suggested needs greater work.

Not really. They're suggesting that you use a different name for the grouping. I'm inclined to agree with them that the current naming is confusing.

**Comment by EvilDragonfiend** [comment]
> > and doing as you suggested needs greater work.
> 
> Not really. They're suggesting that you use a different name for the grouping. I'm inclined to agree with them, the current naming is confusing.

the most issue is that the only condition is if you have R_ADMIN or not in admin group. mentors aren't in GLOB.admin, and they're in GLOB.mentors so they're fine with this issue, but coders are mixed within GLOB.admin.

File: code/modules/client/verbs/who.dm [hunk 2]
```diff
@@ -100,13 +103,30 @@
...
+					admin_list += "\t[C] is \a [rank]\n"
+				else // admins without R_ADMIN perm should be sorted in different area, so that people won't believe coders will handle ahelp
+					var/rank = "\improper [C.holder.rank]"
+					non_admin_list += "\t[C] is \a [rank]\n"
+
+		msg = "<b>Current Admins:</b>\n"
+		for(var/each in admin_list)
+			msg += each
+		if(length(non_admin_list)) // notifying the absence of non-admins has no point
+			msg += "<b>Current Non-admin staffs:</b>\n"
```
Comment (Rukofamicom) [inline code/modules/client/verbs/who.dm]:
```suggestion
			msg += "<b>Current Maintainers:</b>\n"
```

**Review by Rukofamicom — CHANGES_REQUESTED** [review]

File: code/modules/client/verbs/who.dm (hunk 2 above)
Comment (Rukofamicom) [reply code/modules/client/verbs/who.dm]: Currently there are only three branches of staff: 
* Admin Team
* Coding Team
* Mentor Team

There's no reason to label this as "non-admin staff" separate from mentors. It contains exclusively maintainers and one head coder.

**Review by Rukofamicom — APPROVED** [review]
Looks good to me now

File: code/modules/client/verbs/who.dm [hunk 3]
```diff
@@ -100,13 +103,30 @@
...
+				else // admins without R_ADMIN perm should be sorted in different area, so that people won't believe coders will handle ahelp
+					var/rank = "\improper [C.holder.rank]"
+					non_admin_list += "\t[C] is \a [rank]\n"
+
+		msg = "<b>Current Admins:</b>\n"
+		for(var/each in admin_list)
+			msg += each
+		if(length(non_admin_list)) // notifying the absence of non-admins has no point
+			msg += "<b>Current Maintainers:</b>\n"
+			msg += "\t(Note: non-admin staff are unable to handle adminhelp tickets.)\n"
```
Comment (itsmeow) [inline code/modules/client/verbs/who.dm]:
```suggestion
			msg += "\t<span class='info'>Non-admin staff are unable to handle adminhelp tickets.</span>\n"
```

**Review by itsmeow — APPROVED** [review]

**Pull request merged**